### ISSUE TITLE
Create Section Header component

### ISSUE THREAD

**Issue by dylan-sftwr** [cerner/terra-core#1270]
# Issue Description
Create a new Section Header component with OCS theming.

## Issue Type

- [x] New Feature
- [ ] Enhancement
- [ ] Bug
- [ ] Other

**Comment by dylan-sftwr** [comment]
# Technical Design: `terra-section-header`

## Summary:
Create a new package called `terra-section-header` to provide sub-header functionality to.

## Responsiveness
No special considerations needed for this component.

## Accessibility
* The `aria-hidden` attribute will be used to indicate the open/closed state of the section header.

## I18N / Bidirectionality
* This component must be able to be rendered with its mirror-imaged equivalent for rtl locales.

## React Props:
| Prop              | Type   | Default     | Required | Description                                                                                       |
| ----------------- | ------ | ----------- | -------- | ------------------------------------------------------------------------------------------------- |
| `isAnimated`      | bool   | `false`     | false    | When set, will animate the display of children and its toggle icon as it is expanded or collapsed |
| `text`            | string | `''`        | true     | Text to be displayed in the section header                                                        |
| `children`        | node   | `undefined` | false    | If present, children are assumed to be hidden and expandable via the toggle icon                  |
| `isInitiallyOpen` | bool   | `false`     | false    | If set, this first renders as expanded                                                            |

## Themeable Styles:
* accordion icon (via an inlined SVG in css)
* background
* border

## Examples:
### Minimal
```jsx
<SectionHeader text="Minimal Example" />
```

### Expandable Animated
```jsx
<SectionHeader
  text="Expandable Example"
  isAnimated
>
  <p>Hello, World!</p>
  .
  .
  .
</>
```

**Comment by dm123455** [comment]
Talked offline we would add another optional prop is_toggler_present!

**Comment by dylan-sftwr** [comment]
@dm123455 done

**Comment by dm123455** [comment]
Thanks! Can we make font size and text transform to be themable?

**Comment by dm123455** [comment]
+1

**Comment by dm123455** [comment]
@neilpfeiffer can you please take a look?

**Comment by ChaseSinclair** [comment]
Is another prop 'count' necessary as well?

**Comment by bjankord** [comment]
+1 on tech design. We'll want to get @neilpfeiffer to take a look and a +1 from one of the other terra maintainers, @mjhenkes, @ryanthemanuel, or @JakeLaCombe

**Comment by mjhenkes** [comment]
I assume this section header is not intended for lists or tables right? Also is the plan to build this on top of the terra-toggle component? Is this a controlled or uncontrolled component?

**Comment by JakeLaCombe** [comment]
Can we make the text a node instead of a string. You should still be able to use strings just fine, and having it as a node makes it easier to add wrappers around the text for functionality such as text truncation or other possible styling.

**Comment by dylan-sftwr** [comment]
@mjhenkes, I hadn't planned on building this on top of the toggle, as I didn't know how it plays into @neilpfeiffer's plan on how this would end up being expanded on during future enhancements. Looking at its implementation, it does look like the two would share quite a bit of the same functionality. If Neil doesn't see an issue building on top of this, I don't see an issue doing so.

Regarding use cases, I'm still waiting on an updated set of callouts.

This is an uncontrolled component, responsible for maintaining all of its own internal state.

**Comment by JakeLaCombe** [comment]
Any chance we could throw a warning instead of an error, like we do for the time-input? https://github.com/cerner/terra-core/blob/master/packages/terra-time-input/src/TimeInput.jsx#L112

**Comment by JakeLaCombe** [comment]
Given my comments, +1

**Comment by mjhenkes** [comment]
After looking deeper into the section header concept and speaking with @dylan-sftwr. I'd recommend that we remove the toggle-able functionality from this section header and replace it with an 'onclick' callback when the toggle button is clicked. Depending on where this section header would be used different methods of 'toggling' will have be be implemented. For instance in a list the style of toggle-ability that this component is suggesting won't work and being able to toggle sections in a list (accordion) is a main use case for a section header like this.

**Comment by bjankord** [comment]
Talked offline in scrum about creating a presentational component, "Section Header" which uses on `isOpen` prop to control its chevron icon placement and an `onClick` prop to trigger state updates. We also plan to create a "Toggle Section Header" which controls its own state and is uncontrolled by parent state similar to the [terra-toggle-button](https://github.com/cerner/terra-core/blob/master/packages/terra-toggle-button/src/ToggleButton.jsx) component. With this component, you'd still be able to set the initial state similar to the toggle-button, but after that, the toggle section header state would be managed within itself.

**Comment by dylan-sftwr** [comment]
@bjankord @mjhenkes @JakeLaCombe @ChaseSinclair @dm123455 @neilpfeiffer,

Can I get eyes on this, and new +1's?

I've updated the tech-design to reflect the decisions made by this morning's post-scrum (this is the purely-presentational implementation, I'll be making a separate issue/tech-design after this component has been implemented, as I plan on consuming it to create the self-contained toggleable version.

**Comment by mjhenkes** [comment]
`isToggleable` could be enabled by the presence of the `onclick` function.
Do we still need `isOpen` when we have `isInitiallyOpen`?

**Comment by dylan-sftwr** [comment]
@mhemesath updated

**Comment by mjhenkes** [comment]
+1

**Comment by ChaseSinclair** [comment]
+1

**Comment by bjankord** [comment]
+1 on tech design.

**Comment by neilpfeiffer** [comment]
+1

**Comment by dm123455** [comment]
+1

**Comment by PaulSolDev** [comment]
JIRA created
